### PR TITLE
add lightwave

### PR DESCRIPTION
**Pull request by max246**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4812

## Checklist:

  - [ x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/light/lightwaverf.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
+LightWaveRF Switch
+==================
+
+.. seo::
+    :description: LightWaveRF Switch Lights
+    :image: brightness-medium.svg
+
+The ``LightWaveRF`` light platform creates a module to dump and send commands to light switches
+:ref:`float output component <output>`.
```
Comment (kbx81) [inline components/light/lightwaverf.rst]: You're referring to an output component here but I don't see that anywhere in the configuration. Is this a copy/paste error?

File: components/light/lightwaverf.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+.. figure:: images/lightwaverf.jpg
+    :align: center
+    :width: 40.0%
+
+LightwaveRF switches are very common in UK automation which allows to control light/socket/relay and more via RF remote
+or via the HUB.
+Using cheap RF receiver and transmitter you can implemented a simple implementation to control your devices.
```
Comment (kbx81) [inline components/light/lightwaverf.rst]:
```suggestion
LightwaveRF switches are very common in UK automation. They allow control of lights, sockets, relays and more via RF remote
 or via a hub.
Using an inexpensive RF transmitter and receiver you can control your devices via ESPHome.
```

File: components/light/lightwaverf.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            code:  [0x04, 0x00, 0x00, 0x00, 0x0f, 0x03, 0x0d, 0x09, 0x08, 0x08]
+            name: "Sofa"
+            repeat: 1
+
+
+Configuration variables module:
+-------------------------------
+
+- **read_pin** (**Required**, int): Internal Pin that the receiver is connected to
+- **write_pin** (**Required**, string): Internal Pin that the transmitter is connected to
```
Comment (kbx81) [inline components/light/lightwaverf.rst]: These are not `int` or `string` parameters. They refer to pin schemas.

```suggestion
- **read_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): The pin that the receiver is connected to
- **write_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): the pin that the transmitter is connected to
```

File: components/light/lightwaverf.rst [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        id: light_off_ceiling_sofa
+        turn_on_action:
+          lightwaverf.send_raw:
+            code:  [0x04, 0x00, 0x00, 0x00, 0x0f, 0x03, 0x0d, 0x09, 0x08, 0x08]
+            name: "Sofa"
+            repeat: 1
+
+
+Configuration variables module:
+-------------------------------
```
Comment (kbx81) [inline components/light/lightwaverf.rst]:
```suggestion
Configuration variables:
------------------------
```

File: components/light/lightwaverf.rst [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+- **read_pin** (**Required**, int): Internal Pin that the receiver is connected to
+- **write_pin** (**Required**, string): Internal Pin that the transmitter is connected to
+
+Configuration variables:
+------------------------
+
+- **name** (*Optional*, string): The name to give for the action
+- **id** (**Required**, :ref:`config-id`): The id of the switch to call via HA
```
Comment (kbx81) [inline components/light/lightwaverf.rst]: I'm not sure what this section is for...can probably delete it

File: components/light/lightwaverf.rst [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            repeat: 1
+
+Configuration variables:
+------------------------
+
+- **name** (*Optional*, string): The name to give for the action
+- **id** (**Required**, :ref:`config-id`): The id of the switch to call via HA
+- **code** (**Required**, list hex): The raw dump in an array of hex
+- **repeat** (*Optional*, int): The amount of repeats to send the signal
+- **invert** (*Optional*, int): Send the signal inverted
```
Comment (kbx81) [inline components/light/lightwaverf.rst]: So it matches what you have in the schema 😉

```suggestion
- **inverted** (*Optional*, boolean): Send the signal inverted
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
Suggestions to fix a few typos and also please double-check that what you have documented matches _exactly_ what is expected by the configuration validation. 🙂

File: components/light/lightwaverf.rst [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+- **id** (**Required**, :ref:`config-id`): The id of the switch to call via HA
+- **code** (**Required**, list hex): The raw dump in an array of hex
+- **repeat** (*Optional*, int): The amount of repeats to send the signal
+- **inverted** (*Optional*, boolean): Send the signal inverted
+
+
+Compatible sensors
+------------------
+
+These are the sensors which are been confirmed to work with the current code base, if you use different onces and they work, please report them to the team.
```
Comment (kbx81) [inline components/light/lightwaverf.rst]:
```suggestion
The RF transmitters/receivers listed below have been confirmed to work with the current code base. If you discover others that work, please let us know!
```

File: components/light/lightwaverf.rst [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+- **name** (*Optional*, string): The name to give for the action
+- **id** (**Required**, :ref:`config-id`): The id of the switch to call via HA
+- **code** (**Required**, list hex): The raw dump in an array of hex
+- **repeat** (*Optional*, int): The amount of repeats to send the signal
+- **inverted** (*Optional*, boolean): Send the signal inverted
+
+
+Compatible sensors
+------------------
```
Comment (kbx81) [inline components/light/lightwaverf.rst]:
```suggestion
Compatible Hardware
-------------------
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
Couple more suggestions 😇

File: components/light/lightwaverf.rst [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        - lightwaverf.send_raw:
+            code:  [0x04, 0x00, 0x00, 0x00, 0x0f, 0x03, 0x0d, 0x09, 0x08, 0x08]
+            name: "Sofa"
+            repeat: 1
+
+Configuration variables:
+------------------------
+
+- **name** (*Optional*, string): The name to give for the action
+- **id** (**Required**, :ref:`config-id`): The id of the switch to call via HA
```
Comment (jesserockz) [inline components/light/lightwaverf.rst]: Copy pasted from somewhere?
```suggestion
```

File: components/light/lightwaverf.rst [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+            code:  [0x04, 0x00, 0x00, 0x00, 0x0f, 0x03, 0x0d, 0x09, 0x08, 0x08]
+            name: "Sofa"
+            repeat: 1
+
+Configuration variables:
+------------------------
+
+- **name** (*Optional*, string): The name to give for the action
+- **code** (**Required**, list hex): The raw dump in an array of hex
+- **repeat** (*Optional*, int): The amount of repeats to send the signal
```
Comment (kbx81) [inline components/light/lightwaverf.rst]: This reads strangely. How about this?
```suggestion
- **repeat** (*Optional*, int): The number of times the message will be repeated
```

**Review by kbx81 — CHANGES_REQUESTED** [review]

File: components/light/lightwaverf.rst [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+Compatible receiver:
+
+- RXB6
+
+
+See Also
+--------
+
+- :doc:`/components/light/index`
+- :doc:`/components/output/esp8266_pwm`
```
Comment (jesserockz) [inline components/light/lightwaverf.rst]: I don't think this is really relevant to this component...
```suggestion
```

File: index.rst [hunk 12]
```diff
@@ -540,7 +540,8 @@ Light Components
 
     Tuya Dimmer, components/light/tuya, tuya.png
     Shelly Dimmer, components/light/shelly_dimmer, shellydimmer2.jpg
-    Custom Light, components/light/custom, language-cpp.svg, dark-invert
+    Custom Light, components/light/custom, language-cpp.svg
```
Comment (jesserockz) [inline index.rst]:
```suggestion
    Custom Light, components/light/custom, language-cpp.svg, dark-invert
```

**Review by jesserockz — APPROVED** [review]

**Pull request merged**